src/Illuminate/Database/Eloquent/Concerns/HasUniqueStringIds.php

```diff
@@ -103,6 +103,6 @@ public function getIncrementing()
      */
     protected function handleInvalidUniqueId($value, $field)
     {
-        throw (new ModelNotFoundException)->setModel(get_class($this), $value);
+        throw (new ModelNotFoundException)->setModel(get_class($this), $value)->setStatus(422);
```

Comment:
Wouldn't it make more sense to throw a different exception (like [`\Ramsey\Uuid\Exception\InvalidUuidStringException`](https://github.com/ramsey/uuid/blob/4.x/src/Exception/InvalidUuidStringException.php))? I mean, the whole point is, that it's not about that the model isn't found.

Reply:
Sharing ModelNotFoundException doesn't seem like the right approach, I agree. Went with a new exception, `InvalidIdFormatException`.

The problem with using `InvalidUuidStringException` is that any time a user requests an endpoint with an invalid UUID, it would be reported (to Sentry, DataDog, et cetera) which isn't desirable.

The user (or the framework) could add it to the exception handler's dontReport array, but there may be times where the user does want it reported in parts of the code outside of route model binding.

It also would mean that every time a user creates their own unique string ID trait, they would have to cook up their own exception. By creating a more generic exception, it saves users from having to follow the additional steps to avoid eating up their exception reporting limits.